### PR TITLE
Utils fixes

### PR DESCRIPTION
**Pull request by Mussuranae**
Some more complicated fixes for the warnings of the utils lib.

### CODE REVIEW

File: libs/media/src/lib/file/explorer/explorer.model.ts [hunk 1]
```diff
@@ -80,22 +81,22 @@ function titlesDirectory(titles: Movie[]) {
 }
 
 function getFormStorage(object: { id: string }, collection: CollectionHoldingFile, label: FileLabel) {
-  const value = getDeepValue(object, getFileMetadata(collection, label, object.id).field);
+  const value = getDeepValue(object, getFileMetadata(collection, label, object.id).field) as Partial<StorageFile>;
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/file/explorer/explorer.model.ts:84]: Maybe a good process should be to specify the output directly as a generic : 
```typescript
function getDeeptValue<T>(....): T
```
Like that we can use it: 
```typescript
const value = getDeepValue<Partial<StorageFile>>(object, getFileMetadata(collection, label, object.id).field)
```

File: apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.html [hunk 2]
```diff
@@ -9,7 +9,7 @@ <h1>My Selection</h1>
       </div>
       <div fxLayout="column" fxLayoutAlign="center center">
         <p *ngIf="bucket | bucketTitles | getTitle | async as titles">
-          {{ (titles | deepKey:'title.international').join(', ') }}
+          {{ titles | deepKey:'title.international' }}
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.html:12]: Is it still working ?

File: libs/media/src/lib/file/upload-widget/upload-widget.component.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class UploadWidgetComponent {
 
     const snapshot = await this.db.doc(docPath).get().toPromise();
     const data = snapshot.data();
-    const media: StorageFile[] | MovieVideo[] | MovieNote[] | string | MovieVideo = getDeepValue(data, field);
+    const media = getDeepValue(data, field) as StorageFile[] | MovieVideo[] | MovieNote[] | string | MovieVideo;
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/file/upload-widget/upload-widget.component.ts:62]: Maybe create a type for that: 
```typescript
type MediaOutput = StorageFile[] | MovieVideo[] | MovieNote[] | string | MovieVideo;
```

File: libs/utils/src/lib/error-logger.module.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ import { MatSnackBar, MatSnackBarModule } from '@angular/material/snack-bar';
 export class ErrorLoggerHandler implements ErrorHandler {
   constructor(private snackBar: MatSnackBar) { }
 
-  handleError(error: any) {
+  handleError(error: unknown) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/error-logger.module.ts:14]: I think it's: 
```typescript
handleError(error: Error | string)
```

File: libs/utils/src/lib/pipes/deep-key.pipe.ts [hunk 5]
```diff
@@ -22,8 +22,11 @@ export class DeepKeyPipe implements PipeTransform {
       const conditions = deepKey.split('||').map(p => p.trim());
 
       if (Array.isArray(value)) {
+        console.log('coucou array')
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/deep-key.pipe.ts:25]: please remove

File: libs/utils/src/lib/pipes/deep-key.pipe.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ import { NgModule } from '@angular/core';
  * const path = 'main.nested.name';
  * this.resolve(result); // 'Joe'
  */
-export const getDeepValue = (object: any, path: string) => path.split('.').reduce((result, key) => result?.[key], object);
+export const getDeepValue = (object: unknown, path: string) => path.split('.').reduce((result, key) => result?.[key], object);
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/deep-key.pipe.ts:13]: I think:
```typescript
export const getDeepValue = (object: Record<string | number, unknown>, path: string)
```

File: apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.html (hunk 2 above)
Comment (Mussuranae) [reply apps/catalog/catalog/src/app/marketplace/title/selection/selection.component.html:12]: Yes, there is the comma, but the space after it has been removed...

**Pull request merged**